### PR TITLE
[No QA] Add deployer GH handle to deploy messages

### PR DESCRIPTION
**Pull request by roryabraham**
@AndrewGable Will you please review this?

### Details
This adds the deployer's GH handle to deploy messages, where the deployer is defined as:

1. For production deploys, the person who closed the checklist to trigger the deploy.
1. For normal staging deploys, the person who merged the PR.
1. For "automatic" CPs (using the `CP Staging` label), the person who merged the PR.
1. For manual CPs using the GH UI, the person who triggered the workflow. (We are going to save this by including the actor's GH handle in the CP branch name.

### Fixed Issues
$ https://github.com/Expensify/Expensify/issues/168264

### Tests
1. Ensure that the deploy checklist is unlocked.
1. Merge this PR, and after the staging deploy completes, verify that it gets a "Deployed to staging by _\<whoever merged it>_" comment.
1. Lock the deploy checklist.
1. Create and merge another PR with the `CP Staging` label. Verify that it gets a "Cherry-picked to staging by _\<whoever merged it>_" comment.
1. Manually CP a PR using the GH UI. For testing purposes, the person who manually CPs the PR should be different than the person who merged it. After the staging deploy completes. verify that it gets a "Cherry-picked to staging by _\<whoever manually triggered the CP>_" comment.

### QA Steps
None.

### Tested On

n/a, GH only. But I did do a _lot_ of incremental testing using a local octokit instance while developing.

### CODE REVIEW

**Review by luacmartins — APPROVED** [review]
Code LGTM! Couldn't test locally, so I'm interested to see how that goes.

File: .github/workflows/cherryPick.yml [hunk 1]
```diff
@@ -128,7 +128,7 @@ jobs:
         # Version: 2.4.3
         uses: repo-sync/pull-request@65194d8015be7624d231796ddee1cd52a5023cb3
         with:
-          source_branch: cherry-pick-staging-${{ github.event.inputs.PULL_REQUEST_NUMBER }}
+          source_branch: ${{ github.actor }}-cherry-pick-staging-${{ github.event.inputs.PULL_REQUEST_NUMBER }}
```
Comment (AndrewGable) [inline .github/workflows/cherryPick.yml:131]: Is there no way to get this or pass this along a different way?

Comment (AndrewGable) [reply .github/workflows/cherryPick.yml:131]: Maybe via the API to check for a workflow https://docs.github.com/en/rest/reference/actions#list-workflow-runs ?

Comment (roryabraham) [reply .github/workflows/cherryPick.yml:131]: I agree this feels hacky, but the difficult part with using this API is identifying the workflow run that's associated with a given PR or version. Let me dig into it a bit.

Comment (roryabraham) [reply .github/workflows/cherryPick.yml:131]: Basically, the information we have when we're running `markPullRequestsAsDeployed` is:

1. The version
2. The PR number

Then given those, we would need to find the `cherryPick.yml` workflow run that:

1. Generated that version, _OR_
1. Was executed with that PR number as input.

But unfortunately the inputs to a workflow run are not saved by their API, so I wasn't able to see how we could identify the correct workflow run.

Comment (roryabraham) [reply .github/workflows/cherryPick.yml:131]: Open to suggestions if you see something I'm missing

Comment (AndrewGable) [reply .github/workflows/cherryPick.yml:131]: I don't see anything on the API docs on first read, but let me keep looking a bit further.

**Review by AndrewGable — APPROVED** [review]

**Comment by AndrewGable** [comment]
Looks great- Feel free to merge when you are ready to test @roryabraham

**Comment by roryabraham** [comment]
Looks like we missed the chance to test this while the current checklist was still unlocked 😞 Gotta test it during the next deploy cycle. @isagoico can you ping or DM me before adding the lock label to the next checklist? Thanks!

**Comment by roryabraham** [comment]
Testing this out now

**Comment by roryabraham** [comment]
Unfortunately it looks like this didn't work as expected https://github.com/Expensify/App/runs/3303693182?check_suite_focus=true

**Comment by roryabraham** [comment]
Okay, just verified over [here](https://github.com/Expensify/App/pull/4593#issuecomment-897975405) that this is fully functional now

**Pull request merged**